tests/unit/services/flash-messages-test.js

```diff
@@ -162,6 +162,24 @@ test('it sets the correct defaults for service properties', function(assert) {
   }
 });
 
+test('when preventDuplicates is `false` setting a message is not required', function(assert) {
+  set(service, 'defaultPreventDuplicates', false);
+
+  service.add({
+    customProperty: 'ohai'
+  });
+
+  assert.equal(get(service, 'queue.0.customProperty'), 'ohai');
+});
+
+test('when preventDuplicates is `true`, setting a message is required', function(assert) {
+  set(service, 'defaultPreventDuplicates', true);
+
+  assert.throws((() => {
+    service.add({ });
```

Comment:
I think `assert.throws` is the right solution to test this. However, you seem to be getting a false positive.  If you change this line to `service.add({ message: 'foo' })` your test will still pass. I think this is due to the wrapped function.

I think you will want something like this:

```
  assert.throws(() => {
    service.add({ });
  },
  ({ message }) => {
    return message == 'Assertion Failed: The flash message cannot be empty when preventDuplicates is enabled.';
  },
    'Error is thrown'
  );
```
`assert.throws` can take 3 arguments. If you pass it only 2, the second is simply the assertion description, so it was not matching the error text here.

The second argument should be a function that accepts the error object and returns true or false.  Give the above example a try and you should see it passes with no message and fails with one.

Reply:
ah, thanks for catching that. Should all be fixed now :)